lib/ui/proposals/list/interactor/proposals_bloc.dart

```diff
@@ -1,68 +1,94 @@
+  ProposalsBloc(this._getProposalsUseCase, this._fetchProfileUseCase, this._errorHandlerManager) : super(const ProposalsState()) {
     on<_Initial>(_initial);
+    on<_Load>(_loadProposalsForEvent);
   }
 
+  List<DaoData> daos = [];
   FilterStatus filterStatus = FilterStatus.active;
+  /// This value can be modified based on your preferences
+  final int first = 10;
+  late int offset;
 
   Future<void> _initial(_Initial event, Emitter<ProposalsState> emit) async {
     if (!event.refresh) {
       emit(state.copyWith(pageState: PageState.loading));
       filterStatus = event.filterStatus;
     }
 
-    await _fetchAndEmitProposals(emit, filterStatus);
+    await _fetchDaos(event, emit);
   }
 
-  Future<void> _fetchAndEmitProposals(Emitter<ProposalsState> emit, FilterStatus filterStatus) async {
-    // Fetch DAOs
+  Future<void> _fetchDaos(_Initial event, Emitter<ProposalsState> emit) async {
     final Result<ProfileData, HyphaError> profileResult = await _fetchProfileUseCase.run();
 
     if (profileResult.isValue && profileResult.asValue!.value.daos.isNotEmpty) {
       daos = profileResult.asValue!.value.daos;
-      // Fetch Proposals using the fetched DAOs
-      final Result<List<ProposalModel>, HyphaError> proposalsResult =
-          await _getProposalsUseCase.run(GetProposalsUseCaseInput(daos, filterStatus));
-
-      if (proposalsResult.isValue) {
-        // Emit both daos and proposals in one state
-        emit(state.copyWith(
-          pageState: PageState.success,
-          proposals: proposalsResult.asValue!.value,
-        ));
+
+      await _loadProposalsForEvent(event, emit);
+    } else {
+      final HyphaError error = profileResult.isError ? profileResult.asError!.error : HyphaError.api('Failed to retrieve DAOs');
+      await _errorHandlerManager.handlerError(error);
+      emit(state.copyWith(pageState: PageState.failure));
+    }
+  }
+
+  Future<void> _loadProposalsForEvent(dynamic event, Emitter<ProposalsState> emit) async {
+    offset = event is _Initial ? 0 : offset + first;
+
+    final List<DaoData> selectedDaos = (GetIt.instance<FilterProposalsBloc>().selectedDaoIds != null && event is _Load)
+        ? daos.where((dao) => GetIt.instance<FilterProposalsBloc>().selectedDaoIds!.contains(dao.docId)).toList()
+        : daos;
+
+    await _fetchAndEmitProposals(event, emit, selectedDaos, 0);
+  }
+
+  Future<void> _fetchAndEmitProposals(dynamic event, Emitter<ProposalsState> emit, List<DaoData> selectedDaos, int nb) async {
+    final Result<List<ProposalModel>, HyphaError> proposalsResult = await _getProposalsUseCase.run(GetProposalsUseCaseInput(selectedDaos, filterStatus, first, offset));
+
+    if (proposalsResult.isValue) {
+      final List<ProposalModel> proposals = event is _Initial ? proposalsResult.asValue!.value : state.proposals + proposalsResult.asValue!.value;
+
+      final List<int> daoIds = GetIt.instance<FilterProposalsBloc>().selectedDaoIds ?? daos.map((DaoData dao) => dao.docId).toList();
```

Comment:
does it always need to filter by daoID even if all daoIDs are selected? e.g. are there results that aren't even in a dao the user is a member of?

I mean it could be - they could have joined a dao and left,

but I would expect that to be part of the query, not additional code running after the query?

Reply:
The selected daos, when `null`, means that all the daos are selected. The `filterByDao` is for UI purposes: the filter page should display the proposal count for all your daos, while the proposals page will show only the selected dao proposals. And yes, you can see only your daos, and all of this is already part of the existing code, which I haven’t changed.

Regarding the scenario where a user could have joined a dao and left, when the filter is initiated (not during pagination), the daos will be up to date. This ensures that the user only sees active daos they are currently a member of.